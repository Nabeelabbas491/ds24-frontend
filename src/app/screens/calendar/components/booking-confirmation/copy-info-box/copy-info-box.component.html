<ds-input-group>
  <ds-input-group-prefix class="w-full">{{ copyText }}</ds-input-group-prefix>
  <button dsButton="icon">
    <ds-icon [name]="icons.Copy" [size]="24" [color]="iconColors.Neutral300" (click)="copyContent()"></ds-icon>
  </button>
</ds-input-group>
