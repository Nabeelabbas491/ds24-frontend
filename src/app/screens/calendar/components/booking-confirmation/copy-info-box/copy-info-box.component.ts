import { Component, Input } from '@angular/core';
import { IconColor, IconName } from '@ds24/elements';

@Component({
  selector: 'ds-copy-info-box',
  templateUrl: './copy-info-box.component.html',
})
export class CopyInfoBoxComponent {
  icons: typeof IconName = IconName;
  iconColors: typeof IconColor = IconColor;

  @Input()
  copyText: string | undefined = '';

  copyContent() {
    //to be implemented
  }
}
