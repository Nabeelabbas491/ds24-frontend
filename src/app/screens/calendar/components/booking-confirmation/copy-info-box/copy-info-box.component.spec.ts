import { ComponentFixture, TestBed } from '@angular/core/testing';

import { CopyInfoBoxComponent } from './copy-info-box.component';

describe('CopyInfoBoxComponent', () => {
  let component: CopyInfoBoxComponent;
  let fixture: ComponentFixture<CopyInfoBoxComponent>;

  beforeEach(() => {
    TestBed.configureTestingModule({
      declarations: [CopyInfoBoxComponent],
    });
    fixture = TestBed.createComponent(CopyInfoBoxComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should compile', () => {
    fixture.detectChanges();
    expect(component).toMatchSnapshot();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });
});
