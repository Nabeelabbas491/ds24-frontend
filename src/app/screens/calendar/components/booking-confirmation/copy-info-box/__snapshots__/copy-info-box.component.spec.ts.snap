// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CopyInfoBoxComponent should compile 1`] = `
CopyInfoBoxComponent {
  "__ngContext__": 0,
  "copyText": "",
  "iconColors": {
    "Danger300": "fill-danger-e300",
    "Neutral0": "fill-neutral-n0",
    "Neutral200": "fill-neutral-n200",
    "Neutral300": "fill-neutral-n300",
    "Neutral400": "fill-neutral-n400",
    "Neutral500": "fill-neutral-n500",
    "Primary100": "fill-primary-p100",
    "Primary300": "fill-primary-p300-main",
    "Primary400": "fill-primary-p400",
    "Success300": "fill-success-s300",
    "Warning300": "fill-warning-w300",
    "white": "fill-white",
  },
  "icons": {
    "ArrowDown": "arrow-down",
    "ArrowLeft": "arrow-left",
    "ArrowUp": "arrow-up",
    "Calendar": "calendar",
    "Camera": "camera",
    "CaretDown": "caret-down",
    "Checked": "checked",
    "Checkmark": "checkmark",
    "Chevron": "chevron",
    "Clear": "clear",
    "Clock": "clock",
    "CoinHands": "coin-hands",
    "Columns": "columns",
    "Copy": "copy",
    "Desktop": "desktop",
    "DialogClose": "dialog-close",
    "DigicallsOrderSales": "digicalls-order-sales",
    "DragAndDrop": "drag-and-drop",
    "ErrorFilled": "error-filled",
    "ExclamationMark": "exclamation-mark",
    "ExclamationMarkFilled": "exclamation-mark-filled",
    "ExternalLink": "external-link",
    "Eye": "eye",
    "EyeCrossed": "eye-crossed",
    "Faq": "faq",
    "File": "file",
    "FilePlay": "file-play",
    "Folder": "folder",
    "Funnel": "funnel",
    "Gear": "gear",
    "Hamburger": "hamburger",
    "InfoFilled": "info-filled",
    "Link": "link",
    "LinkHorizontal": "link-horizontal",
    "Loader": "loader",
    "MagnifyingGlass": "magnifying-glass",
    "Members": "members",
    "Minus": "minus",
    "Mobile": "mobile",
    "More": "more",
    "Pencil": "pencil",
    "Phone": "phone",
    "Picture": "picture",
    "Plug": "plug",
    "Plus": "plus",
    "Popup": "popup",
    "QuestionMark": "question-mark",
    "Redo": "redo",
    "Restore": "restore",
    "Rows": "rows",
    "Save": "save",
    "Sections": "sections",
    "Skip": "skip",
    "Stack": "stack",
    "Stats": "stats",
    "Tablet": "tablet",
    "Tag": "tag",
    "Text": "text",
    "Translation": "translation",
    "Trash": "trash",
    "Undo": "undo",
    "UserMoney": "user-money",
  },
}
`;
